CMakeLists.txt

```diff
@@ -23,9 +23,10 @@
 #
 #=============================================================================
 
-cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.15.2 FATAL_ERROR)
```

Comment:
This would make pocl unbuildable on every major Linux distribution. At the very least we want it working on the latest Ubuntu LTS (18.04) and Debian (10).

Reply:
I think the "official policy" in pocl we set place was to support the latest Debian and latest Ubuntu LTS with community support, so this includes Ubuntu 16.04, but not anymore 14.04. 

Ubuntu 16.04 ships 3.5.1 and Debian 10 ships 3.13.4. So from that perspective we could bump that to 3.5.1. But again, why? If we don't know of any issue with 2.8.12, why make the life (possibly) harder for some older distro which is stuck with Cmake?

Reply:
I don't remember why. Probably something related to some modules used. I can downgrade the required version though, but I am not sure that it will work on their distros. IMHO  if they have an old CMake, they should just upgrade it. Kitware ships prebuilt binaries. I personnaly have set up a deb repo with a bleeding edge CMake and use it.

Reply:
> I don't remember why. 

OK, then please just revert it so we can move forward with this PR.

Reply:
d94c3a5e647e64876c692a470ba62e5b703735d0